### PR TITLE
provide the ability to run mdsctl from PATH

### PR DESCRIPTION
**Pull request by jwtodd**
mdsctl should be invokable from current-working-directory and, additionally, modifies the PATH env such that subsequent sh sessions work as expected.

test:

- install mds
- add mds-installation-dir to PATH
  - `% export PATH=${MDS_HOME}/bin:${PATH}
- invoke mdsctl from any location
  - `% cd /tmp; mdsctl`

note:

- display installation directory: `% mdsctl --home`
- make mdsctl a-bit-less-verbose: `% mdsctl --quiet ...`

## PR Checklist

 - [ ] simple searchable title - `[mds-db] Add PG env var`, `[config] Fix eslint config`
 - [ ] briefly describe the changes in this PR
 - [ ] mark as draft if should not be merged
 - [ ] write tests for all new functionality

## Impacts
- [ ] Provider
- [ ] Agency
- [ ] Audit
- [ ] Policy
- [ ] Compliance
- [ ] Daily
- [ ] Native
- [ ] Policy Author

### CODE REVIEW

File: bin/mdsctl [hunk 1]
```diff
@@ -127,6 +152,8 @@ options:
   --configure:{key}+={value}                          : append configuration value by key
   --configure:{key}=                                  : clears configuration value by key
   --preset:[preset-key[,preset-key]]                  : preset configurations
+  --home                                              : returns installation directory
+  --quiet                                             : less verbose
```
Comment (drtyh2o) [inline bin/mdsctl]: Controlling verbosity via an environment variable might be nice at some point. I suppose one could also `alias mdsctl=mdsctl --quiet`

Comment (jwtodd) [reply bin/mdsctl]: yeah ... i envision we will want to do something like ssh's verbosity switch (iirc), in that the more v's one provides the more verbose, eg: ... -vvv

**Comment by jwtodd** [comment]
yeah, have done this stuff before: see https://github.com/intuit/wasabi/blob/develop/bin/wasabi.sh#L75

**Comment by avatarneil** [comment]
I'd really like to see mdsctl get _automatically_ added to PATH by updating common `.{bash|zsh|...}rc` files. Any thoughts?

**Comment by jwtodd** [comment]
@avatarneil +1 ... what i have done in the past, and this is a tad tricky, is effectively:

- check if mdsctl is in path (trivial, mdsctl does alot of this already for toolchain dep checks)
- if mdsctl is not in the path append ~/.bashrc with 'export PATH` one-liner, eg:
  - `export PATH=[fully-qualified-mdsctl-install-path]/bin:${PATH}`
- now if mdsctl is in the path but requires a change, eg upgrade, then we need to do an in-file sed replacement

the user would still have to source/dot the change for the existing session to work as expected ... although mdsctl presently does support respawning a subshell and exiting the parent shell today ( see: https://github.com/lacuna-tech/mds-core/blob/develop/bin/mdsctl#L28 )

i can work up a patch to support an ~/.bashrc addition, take on non-bash env (eg zsh) and mdsctl upgrades as follow on, being informed by this initial spike? supporting other shells should just be a ${SHELL} switch statement, me thinks. q: ${SHELL} on zsh works as expected, eg echoing '/bin/bash'?

**Comment by jwtodd** [comment]
@avatarneil or ... we could go all in on brew, create an mdsctl brew recipe and get the env pathing for free. i've created brew recipes before, just need to bundle up the deps (the bin and helm dirs) then post a gzip somewhere ( https://github.com/lacuna-tech/mds-core/releases me thinks).

i <3 this later approach

**Comment by jwtodd** [comment]
in my experience, we need both. and starting with the non-brew solution is immediately helpful to those that have the source cloned locally.

File: bin/mdsctl [hunk 2]
```diff
@@ -65,8 +82,12 @@ declare -A configs=(
...
   ["setfiles-mds"]="${MDS_SET_FILES_MDS:-}"
   ["namespace"]="${MDS_NAMESPACE:-default}"
-  ["namespace-mds"]="${MDS_NAMESPACE_MDS}"
-  ["dryrun"]="${MDS_DRY_RUN:-}")
+  ["namespace-metrics-adapter"]="${MDS_NAMESPACE_METRICS_ADAPTER:-util}"
+  ["namespace-dashboard"]="${MDS_NAMESPACE_LOGGING:-dashboard}"
+  ["namespace-logging"]="${MDS_NAMESPACE_LOGGING:-util}"
+  ["namespace-mds"]="${MDS_NAMESPACE_MDS:-mds}"
+  ["dryrun"]="${MDS_DRY_RUN:-}"
+  ["quiet"]="${MDS_QUIET:-}")
```
Comment (jwtodd) [inline bin/mdsctl]: @drtyh2o an existing-and-non-empty env var MDS_QUIET will effectively inline '--quiet' for every invocation, eg:

`export MDS_QUIET=true`

**Comment by jwtodd** [comment]
@avatarneil this latest patch updates PATH to include mdsctl if it isn't already.

will take on a homebrew distribution subsequently, as it will require a bit of refactoring (namely externalizing brew dependencies).

**Pull request merged**